src/vm/core.zig

```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
         return self.runInstruction(allocator, &instruction);
     }
 
+    pub fn insertDeducedOperands(self: *Self, allocator: Allocator, op: OperandsResult) !void {
+        try self.segments.memory.set(allocator, op.op_0_addr, op.op_0);
```

Comment:
I think we must only do that if op0 was deducted: https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L369

Maybe we can implement the util logic https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L54-L74 in our `OperandsResult` structure wdt?

Reply:
Are u thinking something like this 
https://github.com/oxlime/cairo-zig/blob/99173d5a81ddac28d6538d15609f0a59690432d8/src/vm/core.zig#L835